commit 950afaf321eedea951334f920822f0136a856017
Author: Yuriy Syrota <ysyrota@gmail.com>
Date:   Fri Jul 6 17:36:21 2012 +0300

    Fixed bug #1529 (Copy/paste problem with Clipboard)

#diff --git a/ChangeLog b/ChangeLog
#index f5a2e55..bb1fed1 100644
#--- a/ChangeLog
#+++ b/ChangeLog
#@@ -1,4 +1,8 @@
# 2011-07-06  Ihor Hayvuk <ihayvuk@ics.com>
#+	* lib/Xm/CopyPaste.c:
#+	Fixed bug #1529 (Copy/paste problem with Clipboard)
#+
#+2011-07-06  Ihor Hayvuk <ihayvuk@ics.com>
# 	* lib/Xm/XmRenderT.c:
# 	Fixed bug #1536 (Performance issue: XmListSetPos() is very very slow)
# 
diff --git a/lib/Xm/CutPaste.c b/lib/Xm/CutPaste.c
index bfb9f30..ba27f1a 100644
--- a/lib/Xm/CutPaste.c
+++ b/lib/Xm/CutPaste.c
@@ -43,6 +43,11 @@ static char rcsid[] = "$TOG: CutPaste.c /main/27 1999/05/26 17:42:48 samborn $"
 #include <stdio.h>
 #include <stdlib.h>
 
+#define FIX_1529
+#ifdef FIX_1529
+#include <X11/Xmd.h>		/* for CARD32 */
+#endif
+
 #define XMERROR(key, message)                                            \
     XtErrorMsg (key, "xmClipboardError", "XmToolkitError", message, NULL, NULL)
 
@@ -1052,6 +1057,12 @@ ClipboardFindItem(
     	return ClipboardFail;
     }
 
+#ifdef FIX_1529
+    if( itemid == XM_HEADER_ID && ptr )
+    {
+    	ptr->header.selectionTimestamp = (CARD32)ptr->header.selectionTimestamp;
+    }
+#endif
     return ClipboardSuccess;
 }
 
